### DISCUSSION TITLE
[HxTreeView] Is it possible to to enable node expansion from clicking the item content?

### DISCUSSION THREAD

**Discussion by rysbilinski** [havit/Havit.Blazor#642] [Q&A]
Currently, it seems the only way to expand nodes in the tree by clicking the expander arrow. I would like to be able to expand by clicking (or double clicking) the whole node - same as in a file explorer.

Is there anyway to achieve this?

**Comment by hakenr** [comment]
We previously discussed this concept, covering both `SelectOnExpand` and `ExpandOnSelect` scenarios. You can refer to our earlier discussion here #64.

When using File Explorer in Windows, I noticed that selecting an item (or a node) does not automatically expand the tree, as shown in this image:
![image](https://github.com/havit/Havit.Blazor/assets/12828888/aa13d917-9775-4c82-bcf6-f4ad95dddab2).

Could you explain your reasoning for why it should behave differently? I'm not opposed to the idea, but I would like to understand the justification before considering the addition of a new parameter to make this behavior configurable.

**Reply by rysbilinski** [reply]
Thanks for your response.

You're right, a single click doesn't expand the node. However, a double-click opens the folder.

reason for needing it - currently it's very small area that you can click to open the folder. This makes for a very fiddly user experience and it's not familiar to our users.
